components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/Extensions/CertificateStoreEx.cs

```diff
@@ -4,45 +4,83 @@
 // ------------------------------------------------------------
 
 namespace Microsoft.Azure.IIoT.OpcUa.Protocol {
-    using Microsoft.Azure.IIoT.Module;
     using Microsoft.Azure.IIoT.OpcUa.Protocol.Runtime;
     using Opc.Ua;
-    using System.Net;
-
+    using System;
 
     /// <summary>
     /// Certificate store extensions
     /// </summary>
     public static class CertificateStoreEx {
 
+
         /// <summary>
-        /// Create trust list
+        /// Build the certificates trust list
         /// </summary>
-        /// <param name="certificateStore"></param>
-        /// <returns></returns>
-        public static CertificateTrustList ToCertificateTrustList(this CertificateStore certificateStore) {
-            var certificateTrustList = new CertificateTrustList {
-                StoreType = certificateStore.StoreType,
-                StorePath = certificateStore.StorePath
-            };
+        public static CertificateTrustList BuildCertificateTrustList(
+            this CertificateStore certificateStore,
+            CertificateTrustList certificateTrustList) {
+            if (certificateStore == null) {
+                throw new ArgumentNullException(nameof(certificateStore));
+            }
+
+            if (certificateTrustList == null) {
+                throw new ArgumentNullException(nameof(certificateTrustList));
+            }
+
+            if (certificateTrustList.StorePath != certificateStore.StorePath) {
+                certificateTrustList.StoreType = certificateStore.StoreType;
+                certificateTrustList.StorePath = certificateStore.StorePath;
+            }
 
             return certificateTrustList;
```

Comment:
Does the method need to return anything given that it is modifying `certificateTrustList` itself ? The return values does not seem to be used.

Reply:
right, I'll refactor this